.github/workflows/ci.yml

```diff
@@ -25,11 +25,27 @@ jobs:
       - uses: actions/checkout@v2
       - name: xref
         run: make xref
+
+      - name: Test ranch-2
+        run: make test REBAR_PROFILE=ranch20
+        if: "${{ matrix.otp_vsn == '24.0' }}"
+      - name: Proper ranch-2
+        run: make proper REBAR_PROFILE=ranch20
+        if: "${{ matrix.otp_vsn == '24.0' }}"
+      - name: Cover ranch-2
+        run: make cover REBAR_PROFILE=ranch20
+        if: "${{ matrix.otp_vsn == '24.0' }}"
+
       - name: Test
         run: make test
+        if: "${{ matrix.otp_vsn != '24.0' }}"
       - name: Proper
         run: make proper
+        if: "${{ matrix.otp_vsn != '24.0' }}"
       - name: Cover
         run: make cover
+        if: "${{ matrix.otp_vsn != '24.0' }}"
```

Comment:
I didn't manage to find a better way to use `ranch20` profile for OTP-24 and `test` profile for al lother versions. Any ideas?

Reply:
I could probably be more explicit and add rebar profile to the matrix, so we run all the tests with both ranch versions, but then exclude some unsupported combinations (like `ranch 1.7 + OTP-24` `ranch 2.0 + OTP<21.3`). Maybe no need to run dialyzer on all of the though...
smth like
```yaml
matrix:
    otp_vsn: [20.3, 21.3, 22.3, 23.3, 24.0]
    profile: [test, ranch20]
    exclude:
      - otp_vsn: 24.0
        profile: test
      - otp_vsn: 20.3
        profile: ranch20
...
steps:
  - name: Test
    run: make test REBAR_PROFILE=${{matrix.profile}}
```